core/src/window_service.rs

```diff
@@ -390,6 +412,56 @@ impl WindowService {
             .unwrap()
     }
 
+    fn start_recover_thread(
+        exit: Arc<AtomicBool>,
+        blockstore: Arc<Blockstore>,
+        reed_solomon_cache: Arc<ReedSolomonCache>,
+        erasure_meta_receiver: Receiver<HashMap<ErasureSetId, ErasureRecoverMeta>>,
+        recovered_shred_sender: Sender<Vec<Shred>>,
+    ) -> JoinHandle<()> {
+        Builder::new()
+            .name("solShrRecov".to_string())
```

Comment:
I know this PR is still marked as a draft, but I think one point that we can land on early is who should own this additional thread. An alternative to what is implemented here (new thread in `WindowService`) that I'll throw out is to have `Blockstore` spin up and own this thread. For teardown, we could add a custom `Drop` implementation to `Blockstore` and join the thread there.

I think there would be several benefits to this:
- We could have a `BlockstoreOptions` flag to toggle async on/off if we wanted
   - For example, if we wanted additional testing before making the change, we could still ship the code to default to inline-recovery and then switch it to use async later
- The unit tests in  `Blockstore` could/would now exercise the async path as well; avoids having a difference between unit tests and `solana-validator` code paths (although the difference would be minimal)
   - We'd probably need to make some of the stuff like `MIN_RECOVERY_DELAY_IN_MS` configurable as our unit tests would likely want to always recover immediately to avoid wasting time
- All of the recovery stuff could remain within `Blockstore`; I personally feel that adding this thread to `WindowService` creates a bit of a coupled interface between `WindowService` and `Blockstore`
- Moreso, all the channels for sending things between between insert() and recovery thread would live within blockstore, so we'd avoid chance of some of the pesky errors we've seen where validator will error/hang on shutdown because we dropped a channel too soon or did some operations out of order between multiple services
   - Granted the WindowService threads hold an `Arc<Blockstore>` so in this case the threads will join before `Blockstore` drops, but still, the point of keeping the ends of the channel in one spot remains

All can/should respond, but @behzadnouri and I recently had an [exchange on a similar point](https://github.com/solana-labs/solana/pull/29004) and I would be curious to hear any thoughts from him. Given all the work he has done with around shreds/erause, we'll want his eyes on this review at some point anyways 😄